src/Simulation/Simulators/SparseSimulator/README.md

```diff
@@ -0,0 +1,155 @@
+
+## Emulation
+Currently the simulator only emulates AND and adjoint AND gates. It asserts that the target qubit is |0> before the AND and after the adjoint AND.
+
+## Statistics
+By default, the simulator tracks the total number of qubits used. It also tracks the number of occupied qubits, which is based on whether they are ever in a non-zero state. That is, if one allocates a qubit and applies Z to it, the simulator will still regard it as unoccupied because it wil still be in the |0> state.
+
+## Assertion optimizations
+Z-assertions can be treated as phase/permutation gates, and thus in a Release build, they are added to the queue and not immediately run. This means if such an assertion fails, it might not give an error until much later in the execution. In a Debug build, they are executed immediately (which will slow down execution, but be more helpful for debugging).
+
+
+# Special Operations
+The sparse simulator has some extra C\# Q\# operations for specific purposes.
+
+## Member functions of `SparseSimulator`
+`SetLogging(bool)`: Logging is set to `false` by default; if set to true, the simulator will output the name of each operation as it executes them. 
+
+## Q\# Functions
+
+`GetAmplitudeFromInt(qubit : Qubit[], label : Int)`: Uses `label` as a bitstring to index the qubits in `qubit`. It returns the amplitude of the state with that label, where the value of all other qubits is 0.
+
+`GetAmplitude(qubit : Qubit[], label : BigInt) : Complex` : Same as `GetAmplitudeFromInt`, but uses a BigInt as an index.
+
+`AssertProbBigInt (stateIndex : BigInt, expected : Double, qubits : Microsoft.Quantum.Arithmetic.LittleEndian, tolerance : Double) : Unit`: Asserts that the probability to measure the qubits in `Qubits` and find the value `stateIndex` is within `tolerance` of the expected probability `expected`.
+
+`Sample(register : qubit [])`: This acts like measuring `register` in the Pauli-Z basis, and returns a boolean array of the result of that measurement, except it is non-destructive. 
+
+# Adding Gates
+
+To add a gate to the simulator, you will need to add it to:
+- SparseSimulatorCS/SparseSimulator.cs
+- Native/capi.cpp
+- Native/quantum_state.hpp
+- Native/basic_quantum_state.hpp
+- Native/SparseSimulator.h
+
+Optionally, you may need to add a Q\# and C\# file to the SparseSimulatorCS folder, to create a Q\# operation to call the gate if it does not already exist.
+
+## Quantum State (quantum_state.hpp and basec_quantum_state.hpp)
```

Comment:
Did you mean 
bas**_i_**c_quantum_state.hpp

instead of 
bas**_e_**c_quantum_state.hpp
?

Reply:
Yes. Addressed it in https://github.com/microsoft/qsharp-runtime/pull/769